upgradability/set_code_hash/named_address_store/lib.rs

```diff
@@ -0,0 +1,134 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+#[openbrush::contract]
+mod named_address_store {
+    use ink_prelude::vec::Vec;
+    use ink_prelude::string::String;
+    use ink_env::set_code_hash;
+
+    pub const NEW_STORAGE_KEY: u32 = openbrush::storage_unique_key!(NewStorage);
+
+    #[derive(Default, Debug)]
+    #[openbrush::upgradeable_storage(NEW_STORAGE_KEY)]
+    pub struct NewStorage {
+        addresses: Vec<(AccountId, Option<String>)>,
+        migration_performed: bool,
+    }
+
+    #[derive(Default)]
+    #[ink(storage)]
+    pub struct NamedAddressStore {
+        _old_storage: Vec<AccountId>,
+        storage: NewStorage,
+    }
+
+    impl NamedAddressStore {
+        /// Initializes empty address store
+        #[ink(constructor)]
+        pub fn new() -> Self {
+            Default::default()
+        }
+
+        /// Performs data migration from the old format
+        #[ink(message)]
+        pub fn migrate(&mut self) {
+            // We only want to perform a migration if it has not happened already
+            if self.storage.migration_performed {
+                return;
+            }
+
+            // Remove duplicates
+            self._old_storage.sort();
+            self._old_storage.dedup();
+
+            // Move data to a new format
+            self.storage.addresses = self._old_storage.iter().map(|&x| (x, None)).collect();
+            self._old_storage.clear();
+            self._old_storage.shrink_to_fit();
+            self.storage.migration_performed = true;
+        }
+
+        /// Sets name for a specified address
+        #[ink(message)]
+        pub fn set_name_for_address(&mut self, address: AccountId, name: String) {
+            for entry in &mut self.storage.addresses {
```

Comment:
loops around unbounded collections are generally a no-no in contracts - use a Mapping and access by key (address)